core/src/test/java/org/infinispan/dataconversion/GzipTranscoder.java

```diff
@@ -0,0 +1,31 @@
+package org.infinispan.dataconversion;
+
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_OBJECT;
+import static org.infinispan.dataconversion.Gzip.compress;
+import static org.infinispan.dataconversion.Gzip.decompress;
+
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.dataconversion.OneToManyTranscoder;
+
+/**
+ * @since 12.0
+ */
+public class GzipTranscoder extends OneToManyTranscoder {
+   public static final String APPLICATION_GZIP_TYPE = "application/gzip";
+   public static final MediaType APPLICATION_GZIP = MediaType.fromString(APPLICATION_GZIP_TYPE);
+
+   public GzipTranscoder() {
+      super(APPLICATION_GZIP, APPLICATION_OBJECT);
+   }
+
+   @Override
+   public Object transcode(Object content, MediaType contentType, MediaType destinationType) {
+      if (destinationType.match(APPLICATION_GZIP)) {
+         return compress(content.toString());
```

Comment:
This sounds dangerous @gustavonalle. Since it supports `application/x-java-object`, users will expect it to work for their custom objects as well. But `decompress` always produces a `String`, so users won't be able to get back any object that's not a `String`.

I suggest removing `GzipTranscoder`.

Reply:
This transcoder is not in production

Reply:
changed

Reply:
I don't like the new transcoder implementation either :)

* The "compression" of Java objects assumes that those objects are serializable, which seems wrong. I'd rather not support `application/x-java-object` at all.
* The decompression only works if the destination media type is `application/x-java-object`, even if the input was a `byte[]` in any other (now lost) media type.

[I still think](https://github.com/infinispan/infinispan/pull/8753#pullrequestreview-505754842) that we'll have to add back some kind of encoding that's independent of transcoding eventually, but right now it would be best to remove `GzipTranscoder`.